components/org.wso2.carbon.identity.multi.attribute.login/org.wso2.carbon.identity.multi.attribute.login.resolver.regex/src/main/java/org/wso2/carbon/identity/multi/attribute/login/resolver/regex/RegexResolver.java

```diff
@@ -158,6 +168,116 @@ private void resolveDistinctUsersForClaims(String loginAttribute, List<String> a
+     *
+     * @param claimURI                 Claim URI.
+     * @param loginAttribute           Login attribute.
+     * @param userStoreManager         User store manager.
+     * @return User list.
+     * @throws org.wso2.carbon.user.core.UserStoreException If an error occurred while getting the user list.
+     */
+    private List<User> getUserListAccordingToUserStorePreferenceOrder(String claimURI, String loginAttribute,
+                                                                      IterativeUserStoreManager userStoreManager)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        List<org.wso2.carbon.user.core.common.User> userList = new ArrayList<>();
+        IterativeUserStoreManager currentUserStoreManager = userStoreManager;
+        while (currentUserStoreManager != null) {
+            userList.addAll(currentUserStoreManager.getAbstractUserStoreManager().getUserListWithID(claimURI,
+                    loginAttribute, null));
+            currentUserStoreManager = currentUserStoreManager.nextUserStoreManager();
+        }
+
+        return userList;
+    }
+
+    /**
+     * This method is used to get the user store preference order.
+     *
+     * @return User store preference order.
+     * @throws org.wso2.carbon.user.core.UserStoreException If an error occurred while getting the user store
+     */
+    private List<String> getUserStorePreferenceOrder() throws org.wso2.carbon.user.core.UserStoreException {
+
+        UserMgtContext userMgtContext = UserCoreUtil.getUserMgtContextFromThreadLocal();
+        if (userMgtContext != null) {
+            UserStorePreferenceOrderSupplier<List<String>>
+                    userStorePreferenceSupplier = userMgtContext.getUserStorePreferenceOrderSupplier();
+            if (userStorePreferenceSupplier != null) {
+                List<String> userStorePreferenceOrder = userStorePreferenceSupplier.get();
+                if (userStorePreferenceOrder != null) {
+                    return userStorePreferenceOrder;
+                }
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    /**
+     * This method is used to generate a user store chain using the user store preference order.
+     *
+     * @param userStorePreferenceOrder User store preference order.
+     * @param abstractUserStoreManager Abstract user store manager.
+     * @return IterativeUserStoreManager.
+     * @throws org.wso2.carbon.user.core.UserStoreException If an error occurred while generating the user store chain.
+     */
+    private IterativeUserStoreManager generateUserStoreChain(List<String> userStorePreferenceOrder,
+                                                             AbstractUserStoreManager abstractUserStoreManager)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        IterativeUserStoreManager initialUserStoreManager = null;
+        IterativeUserStoreManager prevUserStoreManager = null;
```

Comment:
It takes some time to understand the logic. Can you check changing the `prevUserStoreManager` to `currentUserStoreManager` and `currentUserStoreManager` to `nextUserStoreManager` makes sense?

Reply:
Resolved with dab73e12b9ba1e14b5e97a8b418985c382751698